File: prefixTree.cpp
class Trie {

 private:
     struct TrieNode{
         struct TrieNode* children[26];
         bool isEnd;

     };

     TrieNode* root;
 
public:
    
     
    Trie() {
         root = new TrieNode();
    }
    
    void insert(string word) {
       TrieNode* p = root;
       for(auto c: word){
             if(p->children[c-'a'] == NULL)p->children[c-'a'] = new TrieNode();
             p = p->children[c-'a'];
       }
       p->isEnd = true;
    }
    
    bool search(string word) {
         TrieNode*p = root;
         for(auto c: word){
               if(p->children[c-'a'] == NULL)return false;
               p = p->children[c-'a'];
         }

         return p->isEnd;
    }
    
    bool startsWith(string prefix) {
        TrieNode* p = root;
        for(auto c: prefix){
              if(p->children[c-'a'] == NULL)return false;
              p = p->children[c-'a']; 
        }

        return true;
    }
};

File: prefixTree.py
class Trie:

    class TrieNode:
        def __init__(self):
            self.children = [None] * 26
            self.isEnd = False

    def __init__(self):
        self.root = self.TrieNode()

    def insert(self, word):
        p = self.root
        for c in word:
            index = ord(c) - ord('a')
            if p.children[index] is None:
                p.children[index] = self.TrieNode()
            p = p.children[index]
        p.isEnd = True

    def search(self, word):
        p = self.root
        for c in word:
            index = ord(c) - ord('a')
            if p.children[index] is None:
                return False
            p = p.children[index]
        return p.isEnd

    def startsWith(self, prefix):
        p = self.root
        for c in prefix:
            index = ord(c) - ord('a')
            if p.children[index] is None:
                return False
            p = p.children[index]
        return True
